bpf/bpf_sock.c

```diff
@@ -210,10 +210,47 @@ int sock4_xlate(struct bpf_sock_addr *ctx)
 	if (!sock_proto_enabled(ctx))
 		return CONNECT_PROCEED;
 
+#ifdef ENABLE_K8S_EXTERNAL_IP  /* ENABLE_K8S_EXTERNAL_IP */
```

Comment:
Is there a way in the daemon where I'd have a --enable-external-ip flag to switch it on/off?

If required for nodeport, it could be enabled transparently, but it would at least give an option to opt-out if the additional overhead would not be needed and a way to debug potential nodeport/external IP separately.

Lets also add that flag to the doc then.

Reply:
Good point, will do this.